test/python/transpiler/test_matrix_utils.py

```diff
@@ -0,0 +1,77 @@
+"""Test matrix utils"""
+
+import unittest
+import numpy as np
+
+from qiskit.test import QiskitTestCase
+from qiskit.transpiler.synthesis.matrix_utils import (
+    build_random_parity_matrix,
+    switch_random_rows,
+    add_random_rows,
+)
+
+
+class TestMatrixUtils(QiskitTestCase):
+    """Test matrix utils"""
+
+    def setUp(self):
+        super().setUp()
+        self.n = np.random.randint(3, 21)
+        self.m = self.n * 10
+        self.matrix = build_random_parity_matrix(self.n, self.m)
+        self.id_matrix = np.identity(self.n)
+
+    def test_build_random_parity_matrix_returns_np_ndarray(self):
+        """Test the output type of build_random_parity_matrix"""
+
+        self.assertIsInstance(self.matrix, np.ndarray)
+
+    def test_build_random_parity_matrix_returns_np_ndarray_of_given_dimensions(self):
+        """Test the output type of build_random_parity_matrix"""
+
+        self.assertEqual(self.matrix.shape, (self.n, self.n))
+
+    def test_build_random_parity_matrix_returns_an_invertible_matrix(self):
+        """Test build_random_parity_matrix for correctness"""
+
+        inv_matrix = np.linalg.inv(self.matrix)
+
+        self.assertIsInstance(inv_matrix, np.ndarray)
+
+    def test_build_random_parity_matrix_does_not_return_an_identity_matrix_when_row_operations_are_executed(
+        self,
+    ):
+        """Test build_random_parity_matrix for correctness"""
+
+        self.assertEqual(np.array_equal(self.matrix, self.id_matrix), False)
```

Comment:
I was wondering does this test fail sometimes if the randomly selected row operations happen to cancel themselves in the end and the result is identity matrix by chance? If that happens, maybe the tests below testing that the matrix changes after switching or adding rows once are enough

Reply:
Good point! Maybe the function itself should have something like an argument `identity_allowed=bool` and if it's false, the function will check if the matrix is an identity matrix, and then perform one more row operation?

Reply:
Or, wouldn't it also prevent the identity matrix from accidentally forming if m is always uneven?

Reply:
Probably, but I'm not sure if that would still work every time because row change can also be done by adding rows together, and maybe that could lead to even number of operations in some cases? In that case the flag could be more secure